src/profile.ts

```diff
@@ -128,31 +141,52 @@ export class Profile {
     profileType: ProfileType,
     config: Config
   ): Promise<IProfile> {
+    let type = AddressType.EOA;
+    let org: Org | null = null;
     const ens = await resolveEnsProfile(addressOrName, profileType, config);
 
     if (ens) {
       if (ens.address) {
+        type = await identifyAddress(ens.address, config);
+
+        if (type === AddressType.Org) {
+          org = await Org.get(ens.address, config);
```

Comment:
The nicest way I found for Orgs is to be part of the Profile type as a kind of subset, to allow us to query for orgs and get all the properties and methods linked to.